### PR TITLE
corefile: expose address to argc

### PR DESCRIPTION
**Pull request by Mic92**
This is useful to reconstruct register states of main even
when the stack is smashed.
As the value is already calculated this does not add much code.

I was lazy and did not run the tests yet, but the code should be covered by the existing doctest already and a quick manual test worked as expected, so please wait for travis to finish.

### CODE REVIEW

**Comment by Mic92** [comment]
Looks like elftools break compatibility: https://travis-ci.com/Mic92/pwntools/builds/86468434#L2101

**Comment by Mic92** [comment]
Ah this was already fixed on master.

**Comment by zachriggle** [comment]
Would you mind extending this to `argv` and `envp` as well?

Thanks!

**Comment by Mic92** [comment]
sure.

File: pwnlib/elf/corefile.py [hunk 1]
```diff
@@ -517,12 +517,21 @@ def __init__(self, *a, **kw):
         #:       address of the string "BAR\x00".
         self.env = {}
 
+        #: :class:`int`: Pointer to envp on the stack
+        self.envp_address = 0
```
Comment (Mic92) [inline pwnlib/elf/corefile.py:521]: `envp_address`, `envp` or `env_address`?

**Comment by Mic92** [comment]
Apart from that it looks ready to me.

**Comment by Mic92** [comment]
Probably my last pull request for quite a while https://github.com/Mic92/pwntools/commit/2b493c64ef758d0f58054c1e9e34055e373ab502

**Pull request merged**